modules/apps/portal-cache/portal-cache-impl/src/main/java/com/liferay/portal/cache/internal/dao/orm/EntityCacheImpl.java

```diff
@@ -230,8 +226,6 @@ public void putResult(
 
 	@Override
 	public void removeCache(String className) {
-		_notifyFinderCache(className, null, true);
-
 		_portalCaches.remove(className);
```

Comment:
@marianoalvarosaiz, removing this from `removeCache(String)` will cause one issue. entityCache.removeCache() will be invoked from ModelPersistenceImpl#deactivate(). Please see the below steps.

1. Go to Web Content, create a web content.
2. Check `com.liferay.journal.model.impl.JournalArticleImpl` related entityCache and finderCache in the cache configuration.
3. Go to gogo shell, find bundleId "com.liferay.journal.service". And then stop it.
Result: Go to cache configuration.  `com.liferay.journal.model.impl.JournalArticleImpl` related entity cache were removed. But finderCache existed. This is not right.

Please add the code back in the `removeCache(String)`. It won't affect the current fix. In addition, I think you may remove the commit message "During the shutdown dispose and/or cache removals are executed and can affect other nodes performance if they are replicated." from the current commit.

Reply:
Thanks @yuhai.

Ok, I see that is used during the deactivation/destroy(spring). I think that with the changes in this pull it doesn't make sense.

The reason why the FinderCache removal is triggered at that point is because of the deactivation of the ModelPersistenceImpl, not by the fact of the removal of the EntityCache. For data modification it makes sense from a conceptual point of view that a change in EntityCache triggers a change in FinderCache, but I think it's best to decouple the removal (that also can be notified across the cluster) of both caches.

This is what I mean: https://github.com/marianoalvarosaiz/liferay-portal/pull/1007 What do you think about this approach?

Reply:
@marianoalvarosaiz, please see https://github.com/marianoalvarosaiz/liferay-portal/pull/1007#issuecomment-1718990476